File: 1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.c
int maxDepth(char* s) {
    char stack[100];
    int top=-1;
    int max=0;
    for(int index=0;s[index]!='\0';index++){
        if(s[index]=='('){
            top++;
            stack[top]='(';
        }
        else if(s[index]==')'){
            top--;
        }

        if(top+1>max){
            max=top+1;
        }
    }
    return max;
}

File: 1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
        
        Stack<Character> stack=new Stack<>();
        int count=0,max=0;
        for(int i=0;i<s.length();i++){

            char ch=s.charAt(i);
            if(ch=='('){
                stack.push(ch);
                count++;
            }
            else if(ch==')'){
                stack.pop();
                count--;
            }
            max=Math.max(max,count);
        }
        
        return max;

    }
}